lib/mox.ex

```diff
@@ -260,25 +248,22 @@ defmodule Mox do
 
   @doc false
   def __dispatch__(mock, name, arity, args) do
-    case Registry.update_value(@name, {self(), mock, name, arity}, &dispatch_update/1) do
+    case Mox.Server.fetch_expectation_for_dispatch({self(), mock, name, arity}) do
       :error ->
         mfa = Exception.format_mfa(mock, name, arity)
         raise UnexpectedCallError, "no expectation defined for #{mfa} in process #{inspect(self())}"
 
-      {_, {count, []}} ->
+      {count, []} ->
```

Comment:
We don't really need to return `{count, []}` from the server. I would say know we can return three direct answers:

  1. `:no_expectation`
  2. `{:out_of_expectations, count}`
  3. `{:ok, fun_to_call}`